internal/nearcache/comparators_test.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nearcache_test
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	inearcache "github.com/hazelcast/hazelcast-go-client/internal/nearcache"
+	"github.com/hazelcast/hazelcast-go-client/nearcache"
+)
+
+type simpleEntry struct {
+	creationTime   int64
+	lastAccessTime int64
+	hits           int64
+}
+
+func (s simpleEntry) Hits() int64 {
+	return s.hits
+}
+
+func (s simpleEntry) Key() interface{} {
+	return nil
+}
+
+func (s simpleEntry) Value() interface{} {
+	return nil
+}
+
+func (s simpleEntry) CreationTime() int64 {
+	return s.creationTime
+}
+
+func (s simpleEntry) LastAccessTime() int64 {
+	return s.lastAccessTime
+}
+
+func TestLRUComparatorDoesNotPrematurelySelectNewlyCreatedEntries(t *testing.T) {
```

Comment:
Can you write where this is ported from as a comment?

Reply:
https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/22d0f4598f4b66079464377bdbe5e4aee0db7a09